services/galley/src/Galley/Cassandra/Conversation.hs

```diff
@@ -266,7 +326,12 @@ toProtocol ::
 toProtocol Nothing _ _ _ = Just ProtocolProteus
 toProtocol (Just ProtocolProteusTag) _ _ _ = Just ProtocolProteus
 toProtocol (Just ProtocolMLSTag) mgid mepoch mcs =
-  ProtocolMLS <$> (ConversationMLSData <$> mgid <*> mepoch <*> mcs)
+  ProtocolMLS <$> (ConversationMLSData <$> mgid <*> mapEpoch mepoch <*> mcs)
+  where
+    -- If there is no epoch in the database, assume the epoch is 0 and this is a
+    -- self-conversation.
+    mapEpoch Nothing = Just (Epoch 0)
+    mapEpoch x = x
```

Comment:
I would just inline `mapEpoch`:

```.haskell
mapEpoch = (<> Just (Epoch 0))
```

Reply:
Done almost the same in commit e2de44d76.